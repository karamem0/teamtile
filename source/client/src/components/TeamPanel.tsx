import React from 'react';
import AppContext from '../contexts/AppContext';
import useTeams from '../hooks/useTeams';
import LoaderPanel from './LoaderPanel';
import EmptyPanel from './EmptyPanel';
import TeamCard from './TeamCard';

const TeamPanel: React.FC = () => {

  const [ token, , , setError ] = React.useContext(AppContext);
  const [ teams, error ] = useTeams({ token });

  React.useEffect(() => {
    if (!setError) {
      return;
    }
    setError(error);
  }, [ setError, error ]);

  if (!teams) {
    return (
      <LoaderPanel />
    );
  }

  if (!teams.length) {
    return (
      <EmptyPanel />
    );
  }

  return (
    <div className="panel panel-grid">
      {
        teams.map((team) =>
          <TeamCard
            key={team.id}
            team={team} />
        )
      }
    </div>
  );

};

export default TeamPanel;
