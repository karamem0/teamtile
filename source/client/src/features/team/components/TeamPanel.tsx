//
// Copyright (c) 2021-2025 karamem0
//
// This software is released under the MIT License.
//
// https://github.com/karamem0/teamtile/blob/main/LICENSE
//

import React from 'react';

import { setFilter, setLoading } from '../../../stores/Action';
import { Event } from '../../../types/Event';
import { clearCache } from '../managers/TeamManager';
import messages from '../messages';
import { useDebounce } from 'react-use';
import { useIntl } from 'react-intl';
import { useStore } from '../../../providers/StoreProvider';

import Presenter from './TeamPanel.presenter';

function TeamPanel() {

  const { dispatch } = useStore();
  const intl = useIntl();
  const [ cache, setCache ] = React.useState<string>();

  useDebounce(() => {
    dispatch(setFilter(cache));
  }, 500, [
    cache,
    dispatch
  ]);

  const handleFilterChange = React.useCallback((_: Event, data?: string) => {
    setCache(data);
  }, []);

  const handleLinkToGitHub = React.useCallback(() => {
    window.open(intl.formatMessage(messages.GitHubLink), '_blank', 'noreferrer');
  }, [
    intl
  ]);

  const handleLinkToPrivacyPolicy = React.useCallback(() => {
    window.open(intl.formatMessage(messages.PrivacyPolicyLink), '_blank', 'noreferrer');
  }, [
    intl
  ]);

  const handleLinkToTermsOfUse = React.useCallback(() => {
    window.open(intl.formatMessage(messages.TermsOfUseLink), '_blank', 'noreferrer');
  }, [
    intl
  ]);

  const handleRefreshClick = React.useCallback(async (event: Event) => {
    const { shiftKey } = event as KeyboardEvent;
    if (shiftKey) {
      await clearCache();
    }
    dispatch(setLoading(true));
  }, [
    dispatch
  ]);

  return (
    <Presenter
      onFilterChange={handleFilterChange}
      onLinkToGitHub={handleLinkToGitHub}
      onLinkToPrivacyPolicy={handleLinkToPrivacyPolicy}
      onLinkToTermsOfUse={handleLinkToTermsOfUse}
      onRefreshClick={handleRefreshClick} />
  );

}

export default TeamPanel;
