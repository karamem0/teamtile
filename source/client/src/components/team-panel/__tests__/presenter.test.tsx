//
// Copyright (c) 2022 karamem0
//
// This software is released under the MIT License.
//
// https://github.com/karamem0/teamtile/blob/main/LICENSE
//

import React from 'react';

import { render, screen } from '@testing-library/react';

import TeamPanel from '../presenter';

beforeEach(() => {
  jest.clearAllMocks();
  jest.restoreAllMocks();
});

describe('TeamPanel', () => {

  it('create shapshot', async () => {
    const params = {
      children: <React.Fragment />,
      filter: undefined
    };
    render(<TeamPanel {...params} />);
    expect(screen.queryAllByText(/^.*$/)[0]).toMatchSnapshot();
  });

});
