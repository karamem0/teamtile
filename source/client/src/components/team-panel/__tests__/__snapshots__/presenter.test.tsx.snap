// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TeamPanel create shapshot 1`] = `
.emotion-0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-flex-flow: column;
  -webkit-flex-flow: column;
  -ms-flex-flow: column;
  flex-flow: column;
  gap: 0.5rem;
}

.emotion-1 {
  display: grid;
  grid-template-rows: auto;
  grid-template-columns: 1fr auto;
  gap: 0.25rem;
}

.emotion-2 {
  min-height: 2.25rem;
}

<body>
  <div>
    <div
      class="emotion-0"
    >
      <div
        class="emotion-1"
      >
        <div
          class="ui-box ui-input emotion-2"
          data-uses-unhanded-props="true"
        >
          <div
            class="ui-box"
            data-uses-unhanded-props="true"
          >
            <input
              class="ui-box ui-input__input"
              data-aa-class="Input__input"
              data-uses-unhanded-props="true"
              id="ui-input-1"
              type="text"
              value=""
            />
            <div
              class="ui-box ui-input__icon"
              data-uses-unhanded-props="true"
            >
              <span
                aria-hidden="true"
                class="root_dd790ee3"
              >
                <svg
                  class="svg_dd790ee3"
                  focusable="false"
                  viewBox="0 0 2048 2048"
                  xmlns="http://www.w3.org/2000/svg"
                >
                  <path
                    d="M1344 0q97 0 187 25t168 71 142 110 111 143 71 168 25 187q0 97-25 187t-71 168-110 142-143 111-168 71-187 25q-125 0-239-42t-211-121l-785 784q-19 19-45 19t-45-19-19-45q0-26 19-45l784-785q-79-96-121-210t-42-240q0-97 25-187t71-168 110-142T989 96t168-71 187-25zm0 1280q119 0 224-45t183-124 123-183 46-224q0-119-45-224t-124-183-183-123-224-46q-119 0-224 45T937 297 814 480t-46 224q0 119 45 224t124 183 183 123 224 46z"
                  />
                </svg>
              </span>
            </div>
          </div>
        </div>
        <button
          class="ui-button"
          data-aa-class="Button"
          data-uses-unhanded-props="true"
        >
          <div
            class="ui-box"
            data-uses-unhanded-props="true"
          >
            <span
              aria-hidden="true"
              class="root_dd790ee3"
            >
              <svg
                class="svg_dd790ee3"
                focusable="false"
                viewBox="0 0 2048 2048"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  d="M1297 38q166 45 304 140t237 226 155 289 55 331q0 141-36 272t-103 245-160 207-208 160-245 103-272 37q-141 0-272-36t-245-103-207-160-160-208-103-244-37-273q0-140 37-272t105-248 167-212 221-164H256V0h512v512H640V215q-117 56-211 140T267 545 164 773t-36 251q0 123 32 237t90 214 141 182 181 140 214 91 238 32q123 0 237-32t214-90 182-141 140-181 91-214 32-238q0-150-48-289t-136-253-207-197-266-124l34-123z"
                />
              </svg>
            </span>
          </div>
        </button>
      </div>
    </div>
  </div>
</body>
`;
